### ISSUE TITLE
开源复刻：用FFmpeg做直播转码集群

### ISSUE THREAD

**Issue by winlinvip** [ossrs/srs#2869]
直播如何实现转多码率，这样PC上看的和手机上看的是不同的流，当然是用FFmpeg，但是需要涉及一些逻辑：

* 从SRS匹配不同的流，并非所有流都要转码。
* 需要有个简单的后台，看到这些转码任务的状态。
* 需要有些模版管理，降低直接用FFmpeg命令行的难度。

如果你需要，请评论区留言。

**Comment by lonelygo** [comment]
Well, this Feather (Demo) is still very meaningful. Transcoding (1 input N output, ABR) is needed for multi-terminal/multi-network environment streaming testing/production.

**Comment by jinleileiking** [comment]
I personally think this is not something SRS should do; it should be done by a larger project (which we are currently doing). Transcoding is a transcoding system; the SRS system and the transcoding system are scheduled through an API: srs-cluster <-- grpc server -> trans system (ffmpegs)

**Comment by winlinvip** [comment]
@jinleileiking This is an open-source replica solution, not transcoding in SRS, but making an additional DEMO, combining FFmpeg and SRS, as well as the management background.

If it's a feature, it wouldn't be called "open-source replica"; it would be called invoking FFmpeg API for transcoding. Calling it an open-source replica means making a complete solution DEMO, and this kind of label is also a Solution, not a Feature.

Many people have implemented live streaming transcoding systems; Zhang San has implemented one, and Li Si has also implemented one, but no one has open-sourced it, and new friends still don't know how to do it. Newcomers can't possibly know how to create a live streaming transcoding system based on just one sentence.

If you're interested, we can work on it together, and there will be a separate repository coming out later.

**Comment by lonelygo** [comment]
Yes, in larger usage scenarios, "Demo" it to let users understand the "best practices" and optional technical routes/solutions for SRS applications in such scenarios.

**Comment by mc373906408** [comment]
我正在尝试用ffmpeg的api全硬件转码H265到H264推流RTMP到SRS

**Comment by mc373906408** [comment]
目前有个很难解决的问题，我推流RTMP，VLC已经能正常播放了，但SRS2RTC很难播放成功，偶尔可以。我使用命令行方式一切正常
